sklearn/sklearn-template/template/README.md

```diff
@@ -0,0 +1,186 @@
+cost-effective, and fully managed cloud data warehouse for analytics, with even 
+built-in machine learning.
+
+## Template structure
+```
+template 
+    |__ config
+        |__ config.yaml             # for running normal training job on AI Platform
+        |__ hptuning_config.yaml    # for running hyperparameter tunning job on AI Platform    
+    |__ scripts
+        |__ train.sh                # convenience script for running machine learning training jobs
+        |__ deploy.sh               # convenience script for deploying trained scikit-learn model
+        |__ predict.sh              # convenience script for requesting online prediction
+        |__ predict.py              # helper function for requesting online prediction using python
+    |__ trainer                     # trainer package
+        |__ metadata.py             # dataset metadata and feature columns definitions
+        |__ model.py                # pre-processing and machine learning model pipeline definition
+        |__ utils.py                # utility functions including e.g. loading data from bigquery and cloud storage
+        |__ task.py                 # training job entry point, handling the parameters passed from command line 
+    |__ setup.py                    # specify necessary dependency for running job on AI Platform
+    |__ requirements.txt            # specify necessary dependency, helper for setup environemnt for local development
+```
+
+## Using the template
+### Step 0. Prerequisites
+Before you follow the instructions below to adapt the tempate to your machine learning job, 
+you need a Google cloud project if you don't have one. You can find detailed instructions 
+[here](https://cloud.google.com/dataproc/docs/guides/setup-project).
+
+- Make sure the following API & Services are enabled.
+    * Cloud Storage
+    * Cloud Machine Learning Engine
+    * BigQuery API
+    * Cloud Build API (for CI/CD integration)
+    * Cloud Source Repositories API (for CI/CD integration)
+
+- Configure project id and bucket id as environment variable.
+  ```bash
+  $ export PROJECT_ID=[your-google-project-id]
+  $ export BUCKET_ID=[your-google-cloud-storage-bucket-name]
+  ```
+  
+- Set up a service account for calls to GCP APIs.  
+  More information on setting up a service account can be found 
+  [here](https://cloud.google.com/docs/authentication/getting-started).
+  
+### Step 1. Tailor the scikit-learn trainer to your data
+
+`metadata.py` is where the dataset's metadata is defined. 
+By default, the file is configured to train on the Iris dataset, which can be found at 
+`bigquery-public-data.ml_datasets.iris`.
+
+```python
+# Example for iris dataset
+CSV_COLUMNS = None  # Schema of the data. Necessary for data stored in GCS
+
+NUMERIC_FEATURES = [
+    'sepal_length',
+    'sepal_width',
+    'petal_length',
+    'petal_width',
+]
+
+# Fill this with any categorical features in the dataset
```

Comment:
I think would be relevant to add a sample using both categorical and numerical in the future.

Reply:
Agreed. I've added a TODO comment in trainer/metadata.py. I'm currently working on getting an ML dataset with categorical variables uploaded in BQ public datasets.